Source/Models/Ratchet/RatchetChat+Encryption.swift

```diff
@@ -0,0 +1,199 @@
+//
+// Copyright (C) 2015-2019 Virgil Security Inc.
+//
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     (1) Redistributions of source code must retain the above copyright
+//     notice, this list of conditions and the following disclaimer.
+//
+//     (2) Redistributions in binary form must reproduce the above copyright
+//     notice, this list of conditions and the following disclaimer in
+//     the documentation and/or other materials provided with the
+//     distribution.
+//
+//     (3) Neither the name of the copyright holder nor the names of its
+//     contributors may be used to endorse or promote products derived from
+//     this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE AUTHOR ''AS IS'' AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+// WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
+// INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+// Lead Maintainer: Virgil Security Inc. <support@virgilsecurity.com>
+//
+
+import VirgilSDKRatchet
+import VirgilCryptoRatchet
+
+extension RatchetChat {
+    /// Encrypts data
+    ///
+    /// - Parameters:
+    ///   - data: Data to encrypt
+    /// - Returns: encrypted Data
+    /// - Throws: corresponding error
+    @objc public func encrypt(data: Data) throws -> Data {
+        let ratchetMessage = try session.encrypt(data: data)
+
+        try self.sessionStorage.storeSession(self.session)
+
+        return ratchetMessage.serialize()
+    }
+
+    /// Decrypts data
+    ///
+    /// - Parameters:
+    ///   - data: encrypted Data
+    /// - Returns: decrypted Data
+    /// - Throws: corresponding error
+    @objc public func decrypt(data: Data) throws -> Data {
+        let message = try RatchetMessage.deserialize(input: data)
+
+        // TODO: Add check on proper session id (local and message one) - should add getter to crypto
```

Comment:
There is not such thing as session id for double ratchet, though decryption will anyway fail in if wrong message is passed

Reply:
Got it. Removed todo